VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/SpigotVoxelSniper.java

```diff
@@ -126,6 +126,8 @@ public void onDisable() {
 
     @EventHandler
     public void onPlayerLeave(PlayerQuitEvent event) {
+        var player = this.players.get(event.getPlayer().getUniqueId());
+        if (player != null) VoxelProfileManager.getInstance().clearSniperForPlayer(player);
```

Comment:
Is it doable to get this handling code in core? That works save us something to forget for forge

Reply:
We probably could, but that would still require some extra code I think (also in the spigot module). I don't think we pass the player leave to the core module at all right now.

Reply:
For now it's fine, something to look into this though

Reply:
We might actually be able to do away with  VoxelProfileManager altogether, we could put the Sniper in the Player class

Reply:
Is the Player class guaranteed to be singleton? If not this wouldn't be safe

Reply:
I kind of made a solution, still requires a bit of code in the spigot module unfortunately